File: app/templates/app-java/src/main/java/ui/main/MainActivity.java
package <%= appPackage %>.ui.main;

import android.os.Bundle;
import <%= appPackage %>.di.ActivityScope;
import <%= appPackage %>.di.components.MainComponent;
import <%= appPackage %>.di.HasComponent;
import <%= appPackage %>.ui.base.BaseActivity;
import <%= appPackage %>.R;
import <%= appPackage %>.application.App;
import <%= appPackage %>.di.components.DaggerMainComponent;
import <%= appPackage %>.di.components.MainComponent;
import <%= appPackage %>.di.modules.MainModule;

import android.os.Bundle;

<% if (nucleus == true) { %>import nucleus.factory.PresenterFactory; <% } %>

import javax.inject.Inject;

@ActivityScope
public class MainActivity extends BaseActivity<MainPresenter> implements MainView, HasComponent<MainComponent> {

        @Inject
        MainPresenter mainPresenter;

        MainComponent component;

        protected void injectModule() {
                component = DaggerMainComponent.builder().applicationComponent(App.graph).mainModule(new MainModule(this)).build();
                component.inject(this);
        }
          <% if (nucleus == true) { %>
        public PresenterFactory<MainPresenter> getPresenterFactory() {
                return () -> mainPresenter;
        }<% } %>

        public void onCreate(Bundle savedInstanceState ) {
                super.onCreate(savedInstanceState);
        }

        protected int getLayoutResource() {
                return R.layout.activity_main;
        }

        @Override
        public MainComponent getComponent() {
          return component;
        }

}

File: app/templates/app-kotlin/src/main/java/ui/main/MainActivity.kt
package <%= appPackage %>.ui.main

import android.os.Bundle
import <%= appPackage %>.di.ActivityScope
import <%= appPackage %>.di.components.MainComponent;
import <%= appPackage %>.di.HasComponent
import <%= appPackage %>.ui.base.BaseActivity
import <%= appPackage %>.R
import <%= appPackage %>.application.App
import <%= appPackage %>.di.components.DaggerMainComponent
import <%= appPackage %>.di.modules.MainModule

<% if (nucleus == true) { %>import nucleus.factory.PresenterFactory <% } %>

import javax.inject.Inject

@ActivityScope
class MainActivity : BaseActivity<MainPresenter>(), MainView, HasComponent<MainComponent>{

    @Inject
    lateinit var mainPresenter: MainPresenter

    lateinit var mainComponent: MainComponent

    override fun injectModule() {
        mainComponent = DaggerMainComponent.builder().applicationComponent(App.get(this).getComponent()).mainModule(MainModule(this)).build()
        mainComponent.inject(this)
    }

    <% if (nucleus == true) { %>override fun getPresenterFactory(): PresenterFactory<MainPresenter>? = PresenterFactory { mainPresenter }<% } else { %>
    override fun getPresenter(): MainPresenter {
        return mainPresenter
    }<% } %>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun getLayoutResource(): Int {
        return R.layout.activity_main
    }

    override fun getComponent(): MainComponent {
        return mainComponent
    }

    <% if (nucleus == false) { %>
    override fun onResume() {
        super.onResume()
        getPresenter().takeView(this)
    }

    override fun onPause() {
        super.onPause()
        getPresenter().dropView()
    }<% } %>
}
